### DISCUSSION TITLE
Load balancer problems

### DISCUSSION THREAD

**Discussion by VA-PLE** [projectcontour/contour#4490] [Q&A] [answered]
Hi. We recently started testing balancing type WeightedLeastRequest(all weights equal). It works good on similar servers. but when the servers are very different the balancing works poorly.
server1 CPU 1x AMD EPYC 7443P 48 cores
server2 CPU 1x Intel® Xeon E-2388G 16 cores
active balancing begins when the second server is heavily overloaded https://ibb.co/hVqfSjS (on the last graph incoming traffic from the Сontour to the worker-node)

please tell me if it can be fixed somehow?

**Comment by youngnick** [comment, marked as answer]
We don't configure Envoy to do any passive load detection on the upstreams, so I don't think it's possible to have Envoy detect this situation automatically.

In this case, I would probably use weights to reflect the increased capacity (since server1 is 4x as big by number of cores), and tweak from there.
